sources/packages/api/src/services/report/report.service.ts

```diff
@@ -0,0 +1,74 @@
+import { Injectable } from "@nestjs/common";
+import { Connection } from "typeorm";
+import { RecordDataModelService } from "../../database/data.model.service";
+import { ReportConfig } from "../../database/entities";
+import { CustomNamedError } from "../../utilities";
+import { ReportsFilterModel } from "./report.models";
+import { REPORT_CONFIG_NOT_FOUND, FILTER_PARAMS_MISMATCH } from "./constants";
+/**
+ * Service layer for reports.
+ */
+@Injectable()
+export class ReportService extends RecordDataModelService<ReportConfig> {
+  constructor(private readonly connection: Connection) {
+    super(connection.getRepository(ReportConfig));
+  }
+
+  /**
+   * Extract the raw data through the dynamic query retrieved from reports config
+   * and translate the dynamic query with arguments built using filter param payload.
+   * @param filter
+   * @returns report raw data.
+   */
+  async exportReport(filter: ReportsFilterModel): Promise<any[]> {
+    const parameters = [];
+    const filterParams = filter.params;
+    const config = await this.getConfig(filter.reportName);
+    if (!config) {
+      throw new CustomNamedError(
+        `The report config is missing for the report ${filter.reportName}.`,
+        REPORT_CONFIG_NOT_FOUND,
+      );
+    }
+    let reportQuery = config.reportSQL;
+
+    Object.keys(filterParams).forEach((key, index) => {
+      const doesParamExist = reportQuery.includes(`:${key}`);
+      if (!doesParamExist) {
```

Comment:
No issues. Because as we discussed in sync up, we are expecting the parameters sent from form.io to match the query arguments(no-optional parameters in front-end). So when there is a mis-match we throw exception.